File: Java/Book_Store/Book.java

package Lab4;

/**
Author : Todd Pidgeon
Date : 03/06/2017
Description : the class structure for book.
**/
public class Book extends Publication {
    private String author;

    public Book(){
        
    }

    public Book(String author) {
        this.author = author;
    }

    public Book(String author, int copies, String title, double price) {
        super(title, price, copies);
        this.author = author;
    }

    public void orderCopies(int numCopies){
        
    }

    /**
     * @return the author
     */
    public String getAuthor() {
        return author;
    }

    /**
     * @param author the author to set
     */
    public void setAuthor(String author) {
        this.author = author;
    }
    
    // A ToString override
    @Override
    public String toString() {
        return "Author : " + getAuthor() + "\n"
            + "Copies : " + getCopies() + "\n"
            + "Title : " + getTitle() + "\n"
            + "Price : " + getPrice() + "\n";
    }
    
    @Override
    public void sellCopy() {
        
    }
}

File: C#/Book_Store_C#/CSD221_Lab6_Todd_Pidgeon/Lab6/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab6
{
    /**
    Author : Todd Pidgeon
    Date : 03/06/2017
    Description : the class structure for book.
    **/
    class Book : Publication
    {
        private String author;

        public String Author
        {
            get { return author; }
            set { author = value; }
        }

        public Book(String title, int copies, double price, String author)
            :base(title, copies, price)
        {
            this.author = author;
        }

        override public void sellCopy()
        {

        }
        override public Double getPrice()
        {
            return Price;
        }

        public override string ToString()
        {
            return "I'm a book";
        }
    }
}
